src/pybind/mgr/dashboard/frontend/src/app/ceph/block/nvmeof-gateway-group/nvmeof-gateway-group.component.ts

```diff
@@ -115,4 +151,52 @@ export class NvmeofGatewayGroupComponent implements OnInit {
   updateSelection(selection: CdTableSelection): void {
     this.selection = selection;
   }
+
+  deleteGatewayGroupModal() {
+    const selectedGroup = this.selection.first();
+    if (!selectedGroup) {
+      return;
+    }
+    const {
+      service_name: serviceName,
+      spec: { group }
+    } = selectedGroup;
+
+    const disable = selectedGroup.subSystemCount > 0 || !group;
+
+    this.modalService.show(DeleteConfirmationModalComponent, {
+      impact: DeletionImpact.high,
+      itemNames: [group],
+      itemDescription: $localize`gateway group`,
+      bodyTemplate: this.deleteTpl,
+      bodyContext: {
+        groupName: group,
+        subsystemCount: selectedGroup.subSystemCount,
+        disable,
+        inputLabel: $localize`Type [resource name] to confirm`,
+        inputPlaceholder: $localize`Name of resource`,
+        customHeader: $localize`Delete gateway group`,
+        customHeaderClass: 'cds--type-heading-compact-01',
+        requireExactMatch: true
```

Comment:
we don't need it. We can rely on `deletionImpact`. if its a high impact deletion, that means it has a textbox to fill in the name and that name should always be equal to the item name. So if we rely on it, the implementation will be generic and all the existing forms can benefit from your implementation.
```suggestion
```